packages/docsearch-react/src/utils/storage.ts

```diff
@@ -82,7 +166,23 @@ export function createObjectStorage<TItem>(key: string) {
       if (item === null) {
         window.localStorage.removeItem(key);
       } else {
-        window.localStorage.setItem(key, JSON.stringify(item));
+        try {
+          window.localStorage.setItem(key, JSON.stringify(item));
+        } catch (error) {
+          // Handle quota exceeded error
+          if (error instanceof DOMException && error.name === 'QuotaExceededError') {
+            try {
+              // First, try comprehensive cleanup of DocSearch storage
+              cleanupDocSearchStorage();
+              // Retry the operation
+              window.localStorage.setItem(key, JSON.stringify(item));
+            } catch {
+              // If still failing, silently fail to prevent crashes
+            }
+          } else {
+            // For other localStorage errors, silently fail to prevent crashes
+          }
+        }
```

Comment:
since it is the same as above, we can extract it in a function, put this in this same file.